perfkitbenchmarker/packages/ycsb.py

```diff
@@ -47,3 +143,533 @@ def YumInstall(vm):
 def AptInstall(vm):
   """Installs the YCSB package on the VM."""
   _Install(vm)
+
+
+def ParseResults(ycsb_result_string, data_type='histogram'):
+  """Parse YCSB results.
+
+  Example input:
+
+    YCSB Client 0.1
+    Command line: -db com.yahoo.ycsb.db.HBaseClient -P /tmp/pkb/workloada
+    [OVERALL], RunTime(ms), 1800413.0
+    [OVERALL], Throughput(ops/sec), 2740.503428935472
+    [UPDATE], Operations, 2468054
+    [UPDATE], AverageLatency(us), 2218.8513395574005
+    [UPDATE], MinLatency(us), 554
+    [UPDATE], MaxLatency(us), 352634
+    [UPDATE], 95thPercentileLatency(ms), 4
+    [UPDATE], 99thPercentileLatency(ms), 7
+    [UPDATE], Return=0, 2468054
+    [UPDATE], 0, 398998
+    [UPDATE], 1, 1015682
+    [UPDATE], 2, 532078
+    ...
+
+  Args:
+    ycsb_result_string: str. Text output from YCSB.
+    data_type: Either 'histogram' or 'timeseries'.
+
+  Returns:
+    A dictionary with keys:
+      client: containing YCSB version information.
+      command_line: Command line executed.
+      groups: list of operation group descriptions, each with schema:
+        group: group name (e.g., update, insert, overall)
+        statistics: dict mapping from statistic name to value
+        histogram: list of (ms_lower_bound, count) tuples, e.g.:
+          [(0, 530), (19, 1)]
+        indicates that 530 ops took between 0ms and 1ms, and 1 took between
+        19ms and 20ms. Empty bins are not reported.
+  """
+  fp = io.BytesIO(ycsb_result_string)
+  client_string = next(fp).strip()
+  if not client_string.startswith('YCSB Client 0.'):
+    raise IOError('Unexpected header: {0}'.format(client_string))
+  command_line = next(fp).strip()
+  if not command_line.startswith('Command line:'):
+    raise IOError('Unexpected second line: {0}'.format(command_line))
+
+  # Some databases print additional output to stdout.
+  # YCSB results start with [<OPERATION_NAME>];
+  # filter to just those lines.
+  def LineFilter(line):
+    return re.search(r'^\[[A-Z]+\]', line) is not None
+  lines = itertools.ifilter(LineFilter, fp)
+
+  r = csv.reader(lines)
+
+  by_operation = itertools.groupby(r, operator.itemgetter(0))
+
+  result = collections.OrderedDict([
+      ('client', client_string),
+      ('command_line', command_line),
+      ('groups', collections.OrderedDict())])
+
+  for operation, lines in by_operation:
+    operation = operation[1:-1].lower()
+    op_result = {
+        'group': operation,
+        data_type: [],
+        'statistics': {}
+    }
+    for _, name, val in lines:
+      name = name.strip()
+      val = val.strip()
+      # Drop ">" from ">1000"
+      if name.startswith('>'):
+        name = name[1:]
+      if name.isdigit():
+        val = float(val) if '.' in val else int(val)
```

Comment:
why not pull this line up above if name.isdigit() and get rid of lines 227-230 below?